master/internal/api_config_policies.go

```diff
@@ -35,29 +43,167 @@ func validatePoliciesAndWorkloadType(workloadType, configPolicies string) error
+}
+
+func (a *apiServer) validateNTSCConfig(
+	ctx context.Context,
+	globalConfigPolicies *model.TaskConfigPolicies,
+	ntscConfigPolicies *model.CommandConfig,
+	constraints *model.Constraints, workloadType string,
+) error {
+	if ntscConfigPolicies.Resources.Priority != nil {
+		if err := a.checkAgainstGlobalPriority(ctx, ntscConfigPolicies.Resources.Priority, workloadType); err != nil {
+			return err
+		}
+	}
+	if err := checkConstraintConflicts(constraints, ntscConfigPolicies.Resources.MaxSlots,
+		&ntscConfigPolicies.Resources.Slots, ntscConfigPolicies.Resources.Priority); err != nil {
+		return err
+	}
+
+	return checkAgainstGlobalConfig(ctx, globalConfigPolicies, nil, ntscConfigPolicies, workloadType)
+}
+
+func (a *apiServer) checkAgainstGlobalPriority(ctx context.Context, taskPriority *int, workloadType string) error {
+	if taskPriority == nil {
+		return nil
+	}
+
+	_, priorityEnabledErr := a.m.rm.SmallerValueIsHigherPriority()
+	if priorityEnabledErr != nil {
+		return fmt.Errorf("task priority is not supported in this cluster: %w", priorityEnabledErr)
+	}
+
+	_, globalPriorityExists, _ := configpolicy.GetPriorityLimit(ctx, nil, workloadType)
+	if globalPriorityExists {
+		return fmt.Errorf(globalPriorityErr)
+	}
+
+	return nil
+}
+
+func checkConstraintConflicts(constraints *model.Constraints, maxSlots, slots, priority *int) error {
+	if constraints == nil {
+		return nil
+	}
+	if priority != nil && *constraints.PriorityLimit != *priority {
+		return fmt.Errorf("invariant config & constraints are trying to set the priority limit")
+	}
+	if maxSlots != nil && *constraints.ResourceConstraints.MaxSlots != *maxSlots {
+		return fmt.Errorf("invariant config & constraints are trying to set the max slots")
+	}
+	if slots != nil && *constraints.ResourceConstraints.MaxSlots > *slots {
+		return fmt.Errorf("invariant config & constraints are attempting to set an invalid max slot")
+	}
+
+	return nil
+}
+
+func checkAgainstGlobalConfig(
+	ctx context.Context, globalConfigPolicies *model.TaskConfigPolicies,
+	expConfig *expconf.ExperimentConfigV0, ntscConfig *model.CommandConfig, workloadType string,
+) error {
+	if globalConfigPolicies == nil || globalConfigPolicies.InvariantConfig == nil {
+		return nil
+	}
+
+	if ntscConfig != nil {
+		globalInvConfig, err := configpolicy.UnmarshalConfigPolicy[model.CommandConfig](
+			*globalConfigPolicies.InvariantConfig, configpolicy.InvalidNTSCConfigPolicyErr,
+		)
+		if err != nil {
+			return err
+		}
+
+		return configpolicy.HaveAtLeastOneSharedDefinedField(globalInvConfig, ntscConfig)
+	}
+
+	if expConfig != nil {
+		globalInvConfig, err := configpolicy.UnmarshalConfigPolicy[expconf.ExperimentConfig](
+			*globalConfigPolicies.InvariantConfig, configpolicy.InvalidExperimentConfigPolicyErr,
+		)
+		if err != nil {
+			return err
+		}
+
+		return configpolicy.HaveAtLeastOneSharedDefinedField(globalInvConfig, expConfig)
```

Comment:
This should also be a warning, not an error. The feature will allow admins to set the same fields at global and workspace level. The workspace policies will not have an effect if a global policy exists for the same field, so we do want to warn them. But, as an example, if an admin wants to switch from global to workspace policies ... they should be allowed to have both set at the same time for an easier migration.